### ISSUE TITLE
Server keeps waiting for crashed AI on startup

### ISSUE THREAD

**Issue by Morlic-fo** [freeorion/freeorion#524]
So I was messing up an import in AI-scripts - AI could not be initialized:

```
2016-03-13 18:37:02.198722 [error] AI : AIClientApp.cpp:51 : AIClientApp::AIClientApp : Failed to initialize Python AI.  Exiting.
2016-03-13 18:37:02.198722 [error] AI : camain.cpp:61 : main() caught exception(std::runtime_error): AIClientApp::AIClientApp : Failed to initialize Python AI.  Exiting.
```

However, the server kept waiting:

```
2016-03-13 18:37:01.244667 [debug] Server : ai-path set to 'AI'
2016-03-13 18:37:01.244667 [debug] Server : ai-config not set.
2016-03-13 18:37:01.248668 [debug] Server : done starting C:\Users\FabianAdmin\FreeOrionDevelopment\FreeOrion\freeorionca.exe
2016-03-13 18:37:01.248668 [debug] Server : (ServerFSM) WaitingForSPGameJoiners.CheckStartConditions
2016-03-13 18:38:13.496800 [debug] Server : ServerNetworking::DisconnectImpl : disconnecting player 1
```

(Last Entry: Me shutting down FO)

Expected Result: Server shuts down if the AI can not be initialized and some error gets displayed.

**Comment by Cjkjvfnby** [comment]
Easy way to reproduce this is to add line `raise Exception('test')` to first line of `FreeOrionAI.py`

**Comment by Cjkjvfnby** [comment]
@LGM-Doyle, If I use `exit(1)` at import stage game will freeze. This code raises `SystemExit`
I don't insist of fixing that.

`v0.4.5+ [build 2016-06-21.94bf367] MSVC 2013`

**Comment by LGM-Doyle** [comment]
@Cjkjvfnby I can replicate this bug.  

The problem is that SystemExit is a python exception that is NOT an error.  Calling PyErr_Print() causes the hang.

I have a fix, but I am done for the day.

It will get fixed but not today.

Can you re-open this issue?  It is a different cause, but the same symptom.

**Comment by Vezzra** [comment]
Fixed by #742
